tests/unit/lms/security_test.py

```diff
@@ -0,0 +1,486 @@
+    def SecurityPolicy(self, patch):
+        return patch("lms.security.SecurityPolicy")
+
+
+class TestTktSecurityPolicy:
+    def test_it_returns_empty_identity_no_cookie(self, pyramid_request):
+        policy = TktSecurityPolicy("TEST_LMS_SECRET")
+
+        identity = policy.identity(pyramid_request)
+
+        assert not identity.userid
+        assert identity.permissions == []
+
+    def test_it_returns_identity_from_cookie(
+        self, pyramid_request, AuthTktCookieHelper
+    ):
+        AuthTktCookieHelper.return_value.identify.return_value = {"userid": "testuser"}
+
+        policy = TktSecurityPolicy("TEST_LMS_SECRET")
+
+        identity = policy.identity(pyramid_request)
+
+        assert identity.userid == "testuser"
+        assert identity.permissions == []
+
+    def test_it_returns_identity_from_cookie_for_reports(
+        self, pyramid_request, AuthTktCookieHelper
+    ):
+        AuthTktCookieHelper.return_value.identify.return_value = {
+            "userid": "report_viewer"
+        }
+
+        policy = TktSecurityPolicy("TEST_LMS_SECRET")
+
+        identity = policy.identity(pyramid_request)
+
+        assert identity.userid == "report_viewer"
+        assert identity.permissions == [Permissions.REPORTS_VIEW]
+
+    def test_it_returns_userid_matches_identity(self, pyramid_request, pyramid_config):
+        pyramid_config.testing_securitypolicy(
+            "testuser", identity=Identity("testuser", [])
+        )
+
+        policy = TktSecurityPolicy("TEST_LMS_SECRET")
+
+        userid = policy.authenticated_userid(pyramid_request)
+
+        assert userid == "testuser"
+
+    def test_permits_allow(self, pyramid_request, pyramid_config):
+        pyramid_config.testing_securitypolicy(
+            "testuser", identity=Identity("testuser", ["some-permission"])
+        )
+
+        policy = TktSecurityPolicy("TEST_LMS_SECRET")
+
+        is_allowed = policy.permits(pyramid_request, None, "some-permission")
+
+        assert is_allowed == Allowed("allowed")
+
+    def test_permits_denied(self, pyramid_request, pyramid_config):
+        pyramid_config.testing_securitypolicy(
+            "testuser", identity=Identity("testuser", [])
+        )
+
+        policy = TktSecurityPolicy("TEST_LMS_SECRET")
+
+        is_allowed = policy.permits(pyramid_request, None, "some-permission")
+
+        assert is_allowed == Denied("denied")
+
+    def test_remember(self, pyramid_request, AuthTktCookieHelper):
+        TktSecurityPolicy("TEST_LMS_SECRET").remember(pyramid_request, "TEST_USERID")
+
+        AuthTktCookieHelper.return_value.remember.assert_called_once()
+
+    def test_forget(self, pyramid_request, AuthTktCookieHelper):
+        TktSecurityPolicy("TEST_LMS_SECRET").forget(pyramid_request)
+
+        AuthTktCookieHelper.return_value.forget.assert_called_once()
+
+    @pytest.fixture
+    def AuthTktCookieHelper(self, patch):
+        return patch("lms.security.AuthTktCookieHelper")
```

Comment:
We should probably try to write the tests without mocking this, though I know the original tests already mocked it

Reply:
You mean setting a cookie on the DummyRequest to get the actual value from the helper?